blocks/amp-story/amp-story-cta-layer.js

```diff
@@ -105,9 +109,25 @@ export default registerBlockType(
 			}
 
 			componentDidMount() {
+				const rootClientID = getBlockRootClientId( this.props.clientId );
+				const blockIndex = wp.data.select( 'core/editor' ).getBlockIndex( rootClientID );
+				let noticeMessage = null;
+				const noticeOptions = {
+					id: 'amp-errors-notice-removed-cta'
+				};
+
 				if ( this.props.attributes.hasMultipleCtaBlocks ) {
 					removeBlock( this.props.clientId );
-					dispatch( 'core/editor' ).createWarningNotice( __( 'Multiple CTA Layers are not allowed, the block was removed.', 'amp' ) );
+					noticeMessage = wp.i18n.__( 'Multiple CTA Layers are not allowed, the block was removed.', 'amp' );
+				}
+
+				if ( 0 === blockIndex ) {
```

Comment:
Maybe this would make sense as `else if ( 0 === blockIndex ) {` ? Since if the block already gets removed within the previous `if` then trying to remove it the second time might cause an error.